app/models/pet.rb

```diff
@@ -13,6 +13,7 @@ def adoption_status
   end
 
   def approved_application
+    return false if application_pets.where(status: "approved").empty?
     self.application_pets.where(status: "approved").first.pet_application
   end
 end
```

Comment:
I am not seeing a test for this model method and created an issue for that to be fixed.